AWS-CodePipeline/two-environment-pipeline-template/{{cookiecutter.outputDir}}/codepipeline.yaml

```diff
@@ -0,0 +1,853 @@
+          - Name: TESTING_IMAGE_REPOSITORY
+            Value: !Ref TestingImageRepository
+          - Name: FEATURE_BRANCH_NAME
+            Value: !Ref FeatureGitBranch
+      ServiceRole: !GetAtt CodeBuildServiceRole.Arn
+      Source:
+        Type: CODEPIPELINE
+        BuildSpec:  |
+          version: 0.2
+          phases:
+            install:
+              runtime-versions:
+                python: 3.8
+              commands:
+                - pip install --upgrade pip
+                - pip install --upgrade awscli aws-sam-cli
+                # Enable docker https://docs.aws.amazon.com/codebuild/latest/userguide/sample-docker-custom-image.html
+                - nohup /usr/local/bin/dockerd --host=unix:///var/run/docker.sock --host=tcp://127.0.0.1:2375 --storage-driver=overlay2 &
+                - timeout 15 sh -c "until docker info; do echo .; sleep 1; done"
+            build:
+              commands:
+                - sam build --use-container --template ${SAM_TEMPLATE}
+                - . ./assume-role.sh ${TESTING_PIPELINE_EXECUTION_ROLE} feature-deploy
+                - sam deploy --stack-name $(echo ${FEATURE_BRANCH_NAME} | tr -cd '[a-zA-Z0-9-]')
+                             --capabilities CAPABILITY_IAM
+                             --region ${TESTING_REGION}
+                             --s3-bucket ${TESTING_ARTIFACT_BUCKET}
+                             {%- if cookiecutter.testing_image_repository %}
+                             --image-repository ${TESTING_IMAGE_REPOSITORY}
+                             {%- endif %}
+                             --no-fail-on-empty-changeset
+                             --role-arn ${TESTING_CLOUDFORMATION_EXECUTION_ROLE}
+
+  CodeBuildProjectBuildAndPackage:
+    Condition: IsMainBranchPipeline
+    Type: AWS::CodeBuild::Project
+    Properties:
+      Artifacts:
+        Type: CODEPIPELINE
+      Environment:
+        Type: LINUX_CONTAINER
+        ComputeType: BUILD_GENERAL1_SMALL
+        Image: aws/codebuild/amazonlinux2-x86_64-standard:3.0
+        PrivilegedMode: true
+        EnvironmentVariables:
+          - Name: SAM_TEMPLATE
+            Value: !Ref SamTemplate
+          - Name: TESTING_REGION
+            Value: !Ref TestingRegion
+          - Name: PROD_REGION
+            Value: !Ref ProdRegion
+          - Name: TESTING_PIPELINE_EXECUTION_ROLE
+            Value: !Ref TestingPipelineExecutionRole
+          - Name: PROD_PIPELINE_EXECUTION_ROLE
+            Value: !Ref ProdPipelineExecutionRole
+          - Name: TESTING_ARTIFACT_BUCKET
+            Value: !Ref TestingArtifactBucket
+          - Name: PROD_ARTIFACT_BUCKET
+            Value: !Ref ProdArtifactBucket
+          - Name: TESTING_IMAGE_REPOSITORY
+            Value: !Ref TestingImageRepository
+          - Name: PROD_IMAGE_REPOSITORY
+            Value: !Ref ProdImageRepository
+      ServiceRole: !GetAtt CodeBuildServiceRole.Arn
+      Source:
+        Type: CODEPIPELINE
+        BuildSpec:  |
```

Comment:
This should really be in a separate file, it's not maintainable as a huge inline string.

Reply:
addressed